pypesto/optimize/ess/ess.py

```diff
@@ -492,43 +490,51 @@ def _do_local_search(
             priority = (
                 1 - self.balance
             ) * quality_order + self.balance * diversity_order
-            chosen_child_idx = np.argmin(priority)
-            local_search_x0 = x_best_children[chosen_child_idx]
-            local_search_fx0 = fx_best_children[chosen_child_idx]
+            local_search_x0_fx0_candidates = (
+                (x_best_children[i], fx_best_children[i])
+                for i in np.argsort(priority)
+            )
```

Comment:
Why change here from "best" to just all of them but sorted?

Reply:
That same point is tried initially. Previously, we would just use the result from that point, even in non-finite. Now, we try the second-best choice if the best one fails (and so on).